service/src/main/java/com/codeforcommunity/database/tableImpl/PostTable.java

```diff
@@ -0,0 +1,153 @@
+    List<SinglePostResponse> posts =
+        Arrays.asList(this.postMap.values().toArray(new SinglePostResponse[0]));
+    return posts;
+  }
+
+  /**
+   * Get the ID of the most recently inserted item. This is so that we can artificially assign a
+   * valid ID to the next item being inserted.
+   *
+   * @return An integer representing the most recent ID.
+   */
+  private int getLastId() {
+    // Get the largest ID available or null.
+    Optional<Integer> maxId = this.postMap.keySet().stream().max(Integer::compareTo);
+    // Return the found ID or -1 if none was found. We want the IDs to be 0 indexed.
+    return maxId.orElse(-1);
+  }
+
+  /**
+   * Create a random date in the form of a string. Static so that we can use it in the {@link
+   * CommentTable} without reimplementing it. Default package access modifier so that only things in
+   * the {@code com.codeforcommunity.tableImpl} package can use it.
+   *
+   * @return A date in the form of a string.
+   */
+  static String getRandDate() {
+    Random random = new Random();
+    // Build a date string. For each number to be inserted, a random number from [0, bound) is
+    // selected, and then is transformed to look more like a date. 1 is added to the day and month
+    // so that you'll have days between [1, 30], and months between [1, 12]. 2000 is added to the
+    // year so you'll get years between 2000 and 2020.
+    String date =
+        String.format(
+            "%d/%d/%d", random.nextInt(12) + 1, random.nextInt(30) + 1, random.nextInt(21) + 2000);
+    return date;
+  }
+
+  /**
+   * Create a random clap count from [0, 1000). Static so that we can use it in the {@link
+   * CommentTable} without reimplementing it. Default package access modifier so that only things in
+   * the {@code com.codeforcommunity.tableImpl} package can use it.
+   *
+   * @return An integer representing a clap count.
+   */
+  static int getRandClapCount() {
+    Random random = new Random();
+    return random.nextInt(1000);
+  }
+
+  @Override
+  public void savePost(CreatePostRequest post) {
+    // Create a new comment using the provided request data, most recently inserted item, and random
+    // data.
+    SinglePostResponse postResponse =
+        new SinglePostResponse(
+            this.getLastId() + 1,
+            post.getAuthor(),
+            getRandDate(),
+            post.getTitle(),
+            getRandClapCount(),
+            post.getBody());
+    this.postMap.put(postResponse.getId(), postResponse);
+  }
+
+  @Override
+  public boolean postExists(int postId) {
+    return this.postMap.containsKey(postId);
+  }
+
+  @Override
+  public void clapPost(int postId) {
+    if (!this.postExists(postId)) {
+      throw new IllegalArgumentException("No post with ID " + postId + " exists");
+    }
+
+    postMap.get(postId).incrementClapCount();
+  }
+
+  @Override
+  public void deletePost(int postId) {
+    postMap.remove(postId);
```

Comment:
The comment table does an exists check before removing, I don't really have a preference either way but lets stay consistent

Reply:
Added.